source/API/CMakeLists.txt

```diff
@@ -185,9 +186,23 @@ add_custom_command_target(
     ALL
     DEPENDS ${clang_headers_target})
 
+add_custom_command_target(
+    unused_var
+    COMMAND "${CMAKE_COMMAND}" "-E" "copy_directory"
+    "${SWIFT_RESOURCE_PATH}" "${lib_dir}/lldb/swift/"
+    OUTPUT "${lib_dir}/lldb/swift/"
```

Comment:
What is this doing?

Reply:
SwiftASTContext::GetResourceDir was assuming that the swift_dir passed in would never be empty, but on Linux it would be empty because we now verify the swift_dir we compute. So I had to copy the swift resource directory into the lldb build tree.

Reply:
I'm not very good at reading CMake: What makes sure this code only runs on Linux?

Reply:
Sorry for the late reply. It doesn't, it copies on every platform. On OSX, the `swift_dir` will get to XCode one, but on other platform, it will have trouble finding the correct swift resource. That's because the `SwiftASTContext::GetResourceDir` makes some assumptions about the resource directory. First, the directory passed in cannot be empty, second if it's a build-script-driven LLDB build, then it will look for sibling directory, either way, we have to provide a valid directory for it.

Reply:
Let me know if I'm misunderstanding something, but that sounds incorrect. On macOS, we distinguish two build schemes, one for local development of LLDB, where copying the Swift resource directory is useful/necessary, and one for building LLDB to ship with Xcode, and in the latter case, we must not copy any resource directories, because LLDB will find the actual Swift compiler's resource directory relative to itself in the Xcode installation.

Reply:
If you look at SwiftASTContext::GetResourceDir (the one that returns a std::string), it tries several strategies to resolve the swift resource directory. If you built lldb using build-script on Linux, then the last strategy in that function is used. That strategy assumes that the parameter `swift_dir` that gets passed in is not empty, even if it's a bogus path. The last strategy currently takes the bogus path of `build/Ninja-DebugAssert/lldb-linux-x86_64/lib/swift` and transforms it into `build/Ninja-DebugAssert/swift-linux-x86_64/lib/swift`.

The problem is that this patch breaks that assumption. The ComputeSwiftResourceDirectory method always verifies that the path to the swift resource directory exists before returning it. This is the same behavior as ComputeClangResourceDirectory. The option that @kusmour went with (that I also think is the right thing to do) is to copy the swift resource directory into the lldb build tree. This ensures that we always will have a swift resource directory in a well-known location on every platform. I can understand if we do not want to install the swift resource directory into a toolchain for distribution because we expect the swift compiler to do it. But for running in the build tree, I think it makes sense.